apps/meteor/client/components/GenericTable/GenericTable.tsx

```diff
@@ -55,14 +55,23 @@ const GenericTable = forwardRef(function GenericTable<
 	const t = useTranslation();
 
 	const [filter, setFilter] = useState(paramsDefault);
+	const [prevSearchTerm, setPrevSearchTerm] = useState<string>('');
 
 	const { itemsPerPage, setItemsPerPage, current, setCurrent, itemsPerPageLabel, showingResultsLabel } = usePagination();
 
 	const params = useDebouncedValue(filter, 500);
 
 	useEffect(() => {
-		setParams({ ...params, text: params.text || '', current, itemsPerPage });
-	}, [params, current, itemsPerPage, setParams]);
+		setParams({
+			...params,
+			text: params.text || '',
+			current: prevSearchTerm === params.text ? current : 0,
+			itemsPerPage,
+		});
+
+		setCurrent(prevSearchTerm === params.text ? current : 0);
+		setPrevSearchTerm(params.text || '');
```

Comment:
do you know that because of that `prevSearchTerm` you are causing an extra render every time? the effect runs 2x
try to create a different effect only if text changes

Reply:
You are right, I followed another approach, thanks!